### PR TITLE
INTMDB-743: [CFN-PI] Update the troubleshooting section in the readme to include common errors

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ INTMDB-743

Several customers are hitting the `CREATE_IN_PROGRESS` issue and we should make sure to highlight this issue and how to fix it. 

Example of customers hitting this issue:
* https://github.com/mongodb/mongodbatlas-cloudformation-resources/issues/506
* https://github.com/aws-quickstart/quickstart-mongodb-atlas/issues/31#issuecomment-1499313698

**This PR update the README to include how to fix this issue.** See [README](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/INTMDB-743#error-the-cfn-stack-remains-in-the-create_in_progress-state-before-failing-after-an-hour-or-so)

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

**Comment by andreaangiolillo** [comment]
Something to consider is if we want a separate `TROUBLESHOOTING.MD` file.

**Review by colm-quinn — APPROVED** [review]
LGTM, also including docs for a review here.

File: README.md [hunk 1]
```diff
@@ -97,3 +97,22 @@ The following are common issues encountered when using AWS CloudFormation/CDK wi
...
+    Statement:
+    - Effect: Allow
+      Principal:
+       Service:
+       - lambda.amazonaws.com
+       - resources.cloudformation.amazonaws.com
+       - cloudformation.amazonaws.com
+      Action: sts:AssumeRole
+```
+We suggest using the [execution-role.yaml](examples/execution-role.yaml) file to generate an IAM role that you can use to activate the CFN resources and run your CFN stacks. Alternatively, you can set the trust relationships of your role through the AWS Console. To do so, go to the IAM dashboard, select your role, and click on Trust Relationships.
```
Comment (jwilliams-mongo) [inline README.md:115]: avoid `we` when you can. The imperative form is more direct.

another minor edit at the end of the line to confirm to our conventions.
```suggestion
Use the [execution-role.yaml](examples/execution-role.yaml) file to generate an IAM role that you can use to activate the CFN resources and run your CFN stacks. Alternatively, you can set the trust relationships of your role through the AWS Console. To do so, go to the IAM dashboard, select your role, and click **Trust Relationships**.
```

**Comment by jwilliams-mongo** [comment]
> LGTM, also including docs for a review here. @jwilliams-mongo is there an alias or process we should use here btw?

Our current process is to file a DOCSP ticket to request a review, but we can refine that if it's not ideal. I'm grabbing this review :)

File: README.md [hunk 2]
```diff
@@ -97,3 +97,22 @@ The following are common issues encountered when using AWS CloudFormation/CDK wi
...
+
+Alternatively, you can set the trust relationships of your role via AWS Console: in the IAM dashboard, select your role and click on `Trust Relationships`:
```
Comment (jwilliams-mongo) [inline README.md:117]: same tweak as in last line in previous paragraph
```suggestion
Alternatively, you can set the trust relationships of your role via AWS Console: in the IAM dashboard, select your role and click **Trust Relationships**:
```

File: README.md [hunk 3]
```diff
@@ -97,3 +97,22 @@ The following are common issues encountered when using AWS CloudFormation/CDK wi
    * When you deploy MongoDB Atlas using CloudFormation with your Atlas PAK, CloudFormation will default to use the IP address of the machine that you are making the API call from. 
    * The machine making the API call to the 3rd-party MongoDB Atlas API would be various AWS servers hosting Lambda functions and won't be static. 
    * Review the [AWS IP address ranges](https://docs.aws.amazon.com/general/latest/gr/aws-ip-ranges.html) and contact AWS Support directly who can help confirm the CIDR range to be used in your Atlas PAK IP Whitelist.
+
+### Error: The CFN stack remains in the `CREATE_IN_PROGRESS` state before failing after an hour or so
+The problem is caused by incorrect trust relationships linked to the role used for activating CFN resources or running the CFN stack. To resolve the issue, ensure that your IAM role's trust relationships include resources.cloudformation.amazonaws.com, cloudformation.amazonaws.com, lambda.amazonaws.com. The following YAML code shows an example of the correct trust relationships:
```
Comment (jwilliams-mongo) [inline README.md:102]: minor edits for clarity, simplifying verb forms, and consistent formatting
```suggestion
The problem is caused by incorrect trust relationships linked to the role that you used to activate CFN resources or run the CFN stack. To resolve the issue, ensure that your IAM role's trust relationships include `resources.cloudformation.amazonaws.com`, `cloudformation.amazonaws.com`, `lambda.amazonaws.com`. The following YAML code shows an example of the correct trust relationships:
```

**Review by jwilliams-mongo — APPROVED** [review]
@andreaangiolillo looks good % minor edits to adhere to docs style

**Comment by andreaangiolillo** [comment]
@jwilliams-mongo Updated. Thanks!

**Review by jwilliams-mongo — APPROVED** [review]
@andreaangiolillo LGTM

**Pull request merged**